tests/unit/test_tutoring_model.py

```diff
@@ -129,6 +129,241 @@ def test_prepare_les_for_ga(learning_style):
+        le = DM.LearningElement(
+            lms_id=i,
+            activity_type="lesson",
+            classification=ele_name,
+            name="Test LE",
+            university="TH-AB",
+            created_by="Max Mustermann",
+            created_at="2023-09-01",
+        )
+        list_of_les3.append(le.serialize())
+    lp = TM.LearningPath(student_id=1, course_id=1, based_on="tyche")
+
+    lp.get_learning_path(
+        student_id=1,
+        learning_style=learning_style,
+        _algorithm="tyche",
+        list_of_les=list_of_les3,
+    )
+    result3 = lp.path
+    erg3 = False
+    if all(le in result3 for le in list_of_keys3):
+        erg3 = True
+    assert erg3
+
+    list_of_les4 = []
+    list_of_keys4 = [
+        "ZF",
+        "RQ",
+        "AN",
+        "EK",
+        "BE",
+    ]
+    for i, ele_name in enumerate(list_of_keys4):
+        le = DM.LearningElement(
+            lms_id=i,
+            activity_type="lesson",
+            classification=ele_name,
+            name="Test LE",
+            university="TH-AB",
+            created_by="Max Mustermann",
+            created_at="2023-09-01",
+        )
+        list_of_les4.append(le.serialize())
+    lp = TM.LearningPath(student_id=1, course_id=1, based_on="tyche")
+
+    lp.get_learning_path(
+        student_id=1,
+        learning_style=learning_style,
+        _algorithm="tyche",
+        list_of_les=list_of_les4,
+    )
+    result4 = lp.path
+    erg4 = False
+    if all(le in result4 for le in list_of_keys4):
+        erg4 = True
+    assert erg4
+
+    # Test invalid error parameter for lp algorithm:
+    with pytest.raises(err.NoValidAlgorithmError):
+        lp.get_learning_path(
+            student_id=1,
+            learning_style=learning_style,
+            _algorithm="foo",
+            list_of_les=list_of_les,
+        )
+
+    # Test Tyche with errors:
+    list_of_les5 = []
+    list_of_keys5 = [
+        "ZF",
+        "KÜ",
+        "SE",
+        "LZ",
+        "ZL",
+        "AN",
+        "ÜB",
+        "EK",
+        "RQ",
+        "FO",
+        "AB",
+        "XX",
+    ]
+    for i, ele_name in enumerate(list_of_keys5):
+        le = DM.LearningElement(
+            lms_id=i,
+            activity_type="lesson",
+            classification=ele_name,
+            name="Test LE",
+            university="TH-AB",
+            created_by="Max Mustermann",
+            created_at="2023-09-01",
+        )
+        list_of_les5.append(le.serialize())
+    with pytest.raises(err.WrongParameterValueError):
+        lp.get_learning_path(
+            student_id=1,
+            learning_style=learning_style,
+            _algorithm="tyche",
+            list_of_les=list_of_les5,
+        )
+    with pytest.raises(err.MissingParameterError):
+        tyche_alg.get_learning_path({}, list_of_les, last_element)
+    with pytest.raises(err.NoValidParameterValueError):
+        tyche_alg.get_learning_path(learning_style, [], last_element)
```

Comment:
same problem with the bool assignment

Reply:
last_element parameter is changed from bool/False to None in tyche.py